odoo/tools/pycompat.py

```diff
@@ -23,8 +24,16 @@ def implements_iterator(cls):
 
     exec ('def reraise(tp, value, tb=None):\n raise tp, value, tb')
 else:
+    import builtins, math
     # pylint: disable=bad-functions
     integer_types = (int,)
+    def round(f):
+        if builtins.round(f + 1) - builtins.round(f) != 1:
+            return f + abs(f) / f * 0.5
+        # In P3, round(-0.) returns 0 rather than -0. so copy sign (which also
+        # ensures the result is a float, as round(v) will return an int
+        # otherwise)
+        return math.copysign(builtins.round(f), f)
```

Comment:
Looks like we're going to call `round()` 3x instead of once in Py3  :cry:  I can't think of any significantly cheaper alternative to the extra `round(f+1)`, but we could store the first `round(f)` and spare the last call, can't we? (seems to be ~20% faster based on a quick microbenchmark)